programs/ssl/ssl_client2.c

```diff
@@ -2178,17 +2421,25 @@ int main( int argc, char *argv[] )
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
     /* The default algorithms profile disables SHA-1, but our tests still
        rely on it heavily. */
+
+#if !defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL) 
```

Comment:
As above: We shouldn't further exacerbate the incompatibility between TLS 1.2 and TLS 1.3 builds. This code path should be enabled even if `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL` is set.